__tests__/utils/ObjectUtils.test.js

```diff
@@ -0,0 +1,83 @@
+import merge from "lodash.merge";
+import { mergeDeepRight, mergeDeepRightObjectArrays } from "../../src/utils/ObjectUtils";
+
+const identifyTraitsPayloadMock = {
+  firstName: "Dummy Name",
+  phone: "1234567890",
+  email: "dummy@email.com",
+  custom_flavor: "chocolate",
+  custom_date: Date.now(),
+  address: [{
+    label: "office",
+    city: "Brussels",
+    country: "Belgium"
+  }, {
+    label: "home",
+    city: "Kolkata",
+    country: "India",
+    nested: {
+      type: "flat"
+    }
+  }, {
+    label: "work",
+    city: "Kolkata",
+    country: "India"
+  }]
+};
+
+const trackTraitsOverridePayloadMock = {
+  address: [{
+    label: "Head office",
+    city: "NYC",
+    country: "Belgium"
```

Comment:
Remove line 32 and the expected merged payload should be the same.

Reply:
did this and added a case for nested object with a property that also has array of objects as value